File: sync/src/main/scala/com/fan/mysql/event/impl/RotateEvent.java
package com.dingcloud.dts.binlog.event.impl;

import com.dingcloud.dts.binlog.event.AbstractEvent;
import com.dingcloud.dts.binlog.event.EventHeader;

public class RotateEvent extends AbstractEvent {

	private long binlogPosition;
	private String binlogFile;

	public RotateEvent(EventHeader header) {
		super(header);
	}
	
	public long getBinlogPosition() {
		return binlogPosition;
	}

	public String getBinlogFile() {
		return binlogFile;
	}

	public void setBinlogPosition(long binlogPosition) {
		this.binlogPosition = binlogPosition;
	}

	public void setBinlogFile(String binlogFile) {
		this.binlogFile = binlogFile;
	}

}

File: async/src/main/scala/com/fan/mysql/async/binlog/event/impl/RotateEvent.scala
package com.fan.mysql.async.binlog.event.impl

import com.fan.mysql.async.binlog.event.{AbstractEvent, EventHeader}

class RotateEvent(header: EventHeader) extends AbstractEvent(header) {
  private[this] var binlogPosition: Long = _
  private[this] var binlogFile: String = _

  def getBinlogPosition: Long = binlogPosition

  def getBinlogFile: String = binlogFile

  def setBinlogPosition(binlogPosition: Long): Unit = {
    this.binlogPosition = binlogPosition
  }

  def setBinlogFile(binlogFile: String): Unit = {
    this.binlogFile = binlogFile
  }
}
